src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -249,16 +249,48 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   const existingPostRun = moduleConfig.postRun || [];
   (moduleConfig as any).preloadPlugins = [];
 
+  var totalResources = Object.keys(resources.assembly).length + Object.keys(resources.pdb || {}).length + /* dotnet.wasm */ 1;
+
+  if (resourceLoader.bootConfig.resources.runtime.hasOwnProperty('dotnet.timezones.blat')){
+    totalResources++;
+  }
+
+  if (resourceLoader.bootConfig.icuDataMode !== ICUDataMode.Invariant){
+    totalResources++;
+  }
+
+  const targetElement = document.getElementById('app') as HTMLDivElement;
+  const progressElement = document.getElementById('progress') as unknown as SVGCircleElement;
+  const percentageElement = document.getElementById('percentage') as unknown as SVGTextElement;
+
+  var resourcesLoaded = 0;
+  function setProgress(){
+      resourcesLoaded++;
+      const percentage = resourcesLoaded / totalResources * 100;
+      
+      if (targetElement) {
+        targetElement.style.setProperty('--blazor-load-percentage', percentage + '%');
+      }
+      
+      if (progressElement && percentageElement) {
+        progressElement.style.display = 'block';
+        percentageElement.textContent = Math.floor(percentage) + '%';
```

Comment:
Rather than hardcoding the idea that there's zero or one elements that will get the percentage applied as text, could we instead define a second CSS variable to hold this text? For example:

```ts
document.documentElement.style.setProperty('--blazor-load-percentage-text', '"' + Math.round(percentage) + '%"');
```

Then the app author is free to use CSS to apply that anywhere in their loading UI if they want, and we wouldn't need to hardcode the IDs `progress` and `percentage`.

Reply:
BTW I'm aware that, with this technique (using an `:after` pseudoelement) to display the text, it will have to go onto a regular `<div>` or similar instead of a `<text>` within the SVG - at least as far as I understand. As far as I can tell, that would be entirely fine and not a drawback.